### PR TITLE
Added field to configure metadata of EO Role/Rolebinding

### PR DESCRIPTION
**Pull request by ShubhamRwt**
Signed-off-by: ShubhamRwt <srawat@redhat.com>

### Type of change

- Enhancement / new feature

### Description

This PR adds field `entityOperatorRoleBinding` and `entityOperatorRole` to configure the metadata of the EntityOperator like lable and annotation.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 * Moved from using the Jaeger exporter to OTLP exporter by default
 * Kafka Exporter support for `Recreate` deployment strategy
 * `ImageStream` validation for Kafka Connect builds on OpenShift
+* Support for configuring the metadata for the Role/Rolebinding of Entity Operator
```
Comment (scholzj) [inline CHANGELOG.md:14]:
```suggestion
* Support for configuring the metadata for the Role / RoleBinding of Entity Operator
```

File: api/src/main/java/io/strimzi/api/kafka/model/template/EntityOperatorTemplate.java [hunk 2]
```diff
@@ -57,6 +58,26 @@ public void setPod(PodTemplate pod) {
...
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public ResourceTemplate getEntityOperatorRole() {
+        return entityOperatorRole;
+    }
+
+    public void setEntityOperatorRole(ResourceTemplate entityOperatorRole) {
+        this.entityOperatorRole = entityOperatorRole;
+    }
+
+    @Description("Template for the Entity operator RoleBinding")
```
Comment (scholzj) [inline api/src/main/java/io/strimzi/api/kafka/model/template/EntityOperatorTemplate.java:71]:
```suggestion
    @Description("Template for the Entity Operator RoleBinding")
```

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/EntityOperatorTest.java [hunk 3]
```diff
@@ -281,6 +284,12 @@ public void testTemplate() {
                                         .withTopologySpreadConstraints(tsc1, tsc2)
                                         .withEnableServiceLinks(false)
                                     .endPod()
+                                    .withNewEntityOperatorRole()
+                                        .withNewMetadata()
+                                            .withLabels(rbLabels)
+                                            .withAnnotations(rbAnots)
+                                        .endMetadata()
+                                    .endEntityOperatorRole()
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/model/EntityOperatorTest.java:292]: Good test. But maybe you can cover the RoleBinding here as well?

File: api/src/main/java/io/strimzi/api/kafka/model/template/EntityOperatorTemplate.java [hunk 4]
```diff
@@ -24,13 +24,14 @@
         builderPackage = Constants.FABRIC8_KUBERNETES_API
 )
 @JsonInclude(JsonInclude.Include.NON_NULL)
-@JsonPropertyOrder({"deployment", "pod", "topicOperatorContainer", "userOperatorContainer", "tlsSidecarContainer", "serviceAccount"})
+@JsonPropertyOrder({"deployment", "pod", "entityOperatorRole", "entityOperatorRoleBinding", "topicOperatorContainer", "userOperatorContainer", "tlsSidecarContainer", "serviceAccount"})
```
Comment (scholzj) [inline api/src/main/java/io/strimzi/api/kafka/model/template/EntityOperatorTemplate.java:27]: Maybe you can just add it at the end? They seem to be the least important.

**Review by scholzj — COMMENTED** [review]
Thanks. I left some nits, but looks mostly good.

File: CHANGELOG.md [hunk 5]
```diff
@@ -11,6 +11,7 @@
 * Moved from using the Jaeger exporter to OTLP exporter by default
 * Kafka Exporter support for `Recreate` deployment strategy
 * `ImageStream` validation for Kafka Connect builds on OpenShift
+* Support for configuring the metadata for the Role/ Rolebinding of Entity Operator
```
Comment (scholzj) [inline CHANGELOG.md:14]:
```suggestion
* Support for configuring the metadata for the Role / RoleBinding of Entity Operator
```

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/EntityOperatorTest.java [hunk 6]
```diff
@@ -233,6 +236,9 @@ public void testTemplate() {
         Map<String, String> saLabels = TestUtils.map("l5", "v5", "l6", "v6");
         Map<String, String> saAnnotations = TestUtils.map("a5", "v5", "a6", "v6");
 
+        Map<String, String> rbLabels = TestUtils.map("l9", "v9", "l10", "v10");
+        Map<String, String> rbAnots = TestUtils.map("a9", "v9", "a10", "v10");
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/model/EntityOperatorTest.java:240]: You should use different set of labels and annos for Role and RolBinding. That way, it tests that you do not mix them as well.

**Review by scholzj — COMMENTED** [review]
Few more nits. Also, looks like you need to update the derived resources to get the CI to pass.

**Review by scholzj — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

**Pull request merged**